src/rgw/rgw_op.cc

```diff
@@ -7438,7 +7443,8 @@ int RGWHandler::do_read_permissions(RGWOp *op, bool only_bucket)
     /* already read bucket info */
     return 0;
   }
-  int ret = rgw_build_object_policies(store, s, op->prefetch_data());
+  bool prefetch_data = op->prefetch_data();
+  int ret = rgw_build_object_policies(store, s, prefetch_data, op->get_range());
```

Comment:
we don't need both `op->prefetch_data()` and `op->get_range()`. can we just have `prefetch_data()` return the pointer?